server/services/echo_service.go

```diff
@@ -46,7 +46,9 @@ func (s *echoServerImpl) Echo(ctx context.Context, in *pb.EchoRequest) (*pb.Echo
 		return nil, err
 	}
 	echoTrailers(ctx)
-	return &pb.EchoResponse{Content: in.GetContent()}, nil
+
+	// bypass the getter for EchoRequest.presence to preserve the exact field value
+	return &pb.EchoResponse{Content: in.GetContent(), Presence: in.Presence}, nil
```

Comment:
just wondering - why `.GetContent()` but just `.Presence`? (you have a getter for `Presence` defined, should we use it?)

Reply:
The comment I placed above this line explains.

The accessor `GetPresence` (see echo.pb.go diff) will return `""` if `Presence` is `nil`. So in order to preserve the _exact_ value, include whether it has been set or not, we must access the field directly, rather than use the `GetPresence` accessor